pkg/network/controllers/vm_test.go

```diff
@@ -608,6 +608,56 @@ var _ = Describe("VM Network Controller", func() {
 		Expect(updatedVMI.Spec.Networks).To(Equal(originalVM.Spec.Template.Spec.Networks))
 		Expect(updatedVMI.Spec.Domain.Devices.Interfaces).To(Equal(originalVM.Spec.Template.Spec.Domain.Devices.Interfaces))
 	})
+
+	It("Could unplug an interface that is pending hotplug", func() {
+		const (
+			netToDetachName    = "detach-me"
+			netToDetachNADName = "detach-me-nad"
+		)
+		clientset := fake.NewSimpleClientset()
+		c := controllers.NewVMController(clientset)
+
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(netToDetachName)),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			libvmi.WithNetwork(libvmi.MultusNetwork(netToDetachName, netToDetachNADName)),
+			libvmistatus.WithStatus(
+				libvmistatus.New(
+					libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+						Name:             defaultNetName,
+						PodInterfaceName: namescheme.PrimaryPodInterfaceName,
+						InfoSource:       vmispec.InfoSourceDomain,
+					}),
+				),
+			),
+		)
+
+		originalVMI := vmi.DeepCopy()
+		vm := libvmi.NewVirtualMachine(vmi.DeepCopy())
+
+		// Simulate the existence of the VMI on the server (to allow the Sync to patch it).
+		_, err := clientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Create(context.Background(), vmi, k8smetav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Mark the secondary interface for hotunplug
+		vm.Spec.Template.Spec.Domain.Devices.Interfaces[1].State = v1.InterfaceStateAbsent
+
+		originalVM := vm.DeepCopy()
+		updatedVM, err := c.Sync(vm, vmi)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(updatedVM.Spec.Template.Spec.Networks).To(Equal(originalVM.Spec.Template.Spec.Networks))
```

Comment:
i am bit confused, the interface is marked absent
so it should be removed with its network because it was unplugged at least according title
(unless it is retained by https://github.com/kubevirt/kubevirt/pull/14914/files#diff-ce73e9866c7d67ddd64b01df072b2914cb040e8cd5795c7abe4803378bb17097R211 because the test need to not match this condition)

Reply:
From the last commit message:
> This is because, the controller does not set the VMI's interface state
as absent, as in this scenario the naming scheme is unknown and assumed
to be ordinal, thus preventing hot-unplug.